risc0/zkvm/src/host/api/mod.rs

```diff
@@ -319,28 +289,45 @@ pub enum AssetRequest {
     Path(PathBuf),
 }
 
+/// Provides information about the result of execution.
+pub struct SessionInfo {
+    /// The number of segments.
+    pub segments: u32,
```

Comment:
Segments seems less useful if the cycle count or po2 for the segments is not known. I may be known by context, if the client requested a specific po2. It would be nice if enough information is included here to additional know (at least an upper bound on) the number of cycles this represents.